File: 14. Longest Common Prefix.js

var longestCommonPrefix = function(strs) {
    if (strs.length === 0)  return "";
    let prefix = strs[0];
    for (let word of strs) {
        while (word.indexOf(prefix) !== 0) {
            prefix = prefix.slice(0, -1);
            if (prefix === '') return ''
        }
    }
    return prefix;
};

File: 14. Longest Common Prefix.kt
class Solution {
    fun longestCommonPrefix(strs: Array<String>): String {
    if (strs.isEmpty()) return ""
    var prefix = strs[0]
    for (word in strs) {
        while (word.indexOf(prefix) != 0) {
            prefix = prefix.slice(0 until prefix.length - 1)
            if (prefix.isEmpty()) return ""
        }
    }
    return prefix
    }
}
